demos/mqtt/mqtt_demo_connection_sharing/mqtt_demo_connection_sharing.c

```diff
@@ -0,0 +1,1601 @@
+                        mqttStatus ) );
+
+            /* Remove registered callback for the notify wildcard topic filter. */
+            ( void ) SubscriptionManager_RemoveCallback( DEMO_HUMIDITY_TOPIC,
+                                                         strlen( DEMO_HUMIDITY_TOPIC ) );
+        }
+    }
+
+    /* PUBLISH to the humidity topic, so that the broker can send the PUBLISH message
+     * back to us. The humidity callback should be invoked on receiving the message back
+     * from the broker, as we have registered the callback for the humidity topic filter. */
+    if( returnStatus == EXIT_SUCCESS )
+    {
+        /* Publish messages to all service topics so that broker will forward it back to us,
+         * and cause registered callbacks to be invoked.  */
+        LogInfo( ( "Publish to topic %s.",
+                   DEMO_HUMIDITY_TOPIC ) );
+
+        returnStatus = publishToTopic( &mqttContext,
+                                       DEMO_HUMIDITY_TOPIC,
+                                       DEMO_HUMIDITY_TOPIC_LENGTH,
+                                       DEMO_HUMIDITY_MESSAGE );
+
+        /* Calling MQTT_ProcessLoop to process incoming publish echo, since
+         * application subscribed to the same topic the broker will send
+         * publish message back to the application. This function also
+         * sends ping request to broker if MQTT_KEEP_ALIVE_INTERVAL_SECONDS
+         * has expired since the last MQTT packet sent and receive
+         * ping responses. */
+        mqttStatus = MQTT_ProcessLoop( &mqttContext, MQTT_PROCESS_LOOP_TIMEOUT_MS );
+
+        if( mqttStatus != MQTTSuccess )
+        {
+            LogWarn( ( "MQTT_ProcessLoop failed: Error = %s.",
+                       MQTT_Status_strerror( mqttStatus ) ) );
+        }
+
+        /* The humidity callback should have been invoked for handling the incoming
+         * PUBLISH message on the humidity topic. */
+        else if( globalReceivedHumidityData != true )
+        {
+            returnStatus = EXIT_FAILURE;
+            LogError( ( "Registered callback was not invoked on sending a PUBLISH to "
+                        "the humidity topic: Topic=%s", DEMO_HUMIDITY_TOPIC ) );
+        }
+    }
+
+    /* Subscribe to the precipitation topic filter, this time also without any wildcard characters,
+     * so that we can receive incoming PUBLISH message only on the same topic from the broker. */
+    if( returnStatus == EXIT_SUCCESS )
+    {
+        /* Register subscription callback for the wildcard notification topic filter. */
+        ( void ) SubscriptionManager_RegisterCallback( DEMO_PRECIPITATION_TOPIC,
+                                                       DEMO_PRECIPITATION_TOPIC_LENGTH,
+                                                       precipitationDataCallback );
+
+        /* The client is now connected to the broker. Subscribe to the precipitation topic
```

Comment:
It's been connected for quite some time now. I'd shorten all subsequent comments after the first one

Reply:
Removed this comment